src/redfish/rest/v1.py

```diff
@@ -307,7 +307,21 @@ def session_location(self):
     @session_location.setter
     def session_location(self, ses_loc):
         """Set the session URI"""
-        self._session_location = ses_loc
+        from urllib.parse import urlparse

```

Comment:
Probably should move the import up to the top.
And I'm not sure if you care about Python2 support? Hopefully not anymore in 2022. But I think the import for Python2 would be different.

Reply:
Yes, Python2 is not supported from version 3.2.0